test/verify/storagelib.py

```diff
@@ -55,6 +55,25 @@ def add_ram_disk(self, size=50):
 
         return dev
 
+    def add_loopback_disk(self, size=50):
+        '''Add per-test loopback disk
+
+        The disk gets removed automatically when the test ends. This is safe for @nondestructive tests.
+
+        Unlike add_ram_disk(), this can be called multiple times, and is less size constrained.
+        However, loopback devices look quite special to the OS, so they are not a very good
+        simulation of a "real" disk.
+
+        Return the device name.
+        '''
+        dev = self.machine.execute("set -e; F=$(mktemp /var/tmp/loop.XXXX); "
+                                   "dd if=/dev/zero of=$F bs=1M count=%s; "
+                                   "losetup --find --show $F; "
+                                   "rm $F" % size).strip()
+        # right after unmounting the device is often still busy, so retry a few times
+        self.addCleanup(self.machine.execute, "umount {0}; until losetup -d {0}; do sleep 1; done".format(dev), timeout=10)
```

Comment:
Wont this run into the same trouble as the `addCleanup` that used to be in `add_ram_disk`?

Reply:
I didn't notice any trouble with that, so as long as it works, it's IMHO more robust to use `addCleanup` at the place where the resource is created. If you prefer, I can move it to `MachineCase.nonDestructiveSetup()` though.

Reply:
No, let's wait for evidence that this is actually problematic.